### PR TITLE
Moved removed events check

### PR DESCRIPTION
**Pull request by Jambrose777**
Moved the removed events check from CAD to the person profile.

Essentially this allows rankings by person for removed events to be visible via manipulating the url (i.e. https://www.worldcubeassociation.org/results/rankings/333ft/single). Currently this doesnt work since the data for person rankings comes from ConciseSingleRanks and ConciseAverageRanks. Whereas other rankings, like by results, work: https://www.worldcubeassociation.org/results/rankings/333ft/single?show=100+results

The only other place I could find where this was affected was the persons profile. These rankings were showing up, so to keep that the same we check the event rank there instead.

I wanted to open this PR to generate discussion about if this is something that we would want to proceed with.

### CODE REVIEW

File: WcaOnRails/app/views/persons/_personal_records.html.erb [hunk 1]
```diff
@@ -18,30 +18,32 @@
...
-              <%= link_to rankings_path(rank_single.event_id, "average"), class: "plain" do %>
-                <%= rank_average&.solve_time&.clock_format %>
-              <% end %>
-            </td>
-            <%= rank_td rank_average, "world" %>
-            <%= rank_td rank_average, "continent" %>
-            <%= rank_td rank_average, "country" %>
-            <td><%= odd_rank_reason if odd_rank_reason_needed?(rank_single, rank_average) %></td>
-          </tr>
+          <% if rank_single.event.rank < 990 %>
```
Comment (jonatanklosko) [inline WcaOnRails/app/views/persons/_personal_records.html.erb:21]: There's a dedicated [`official?`](https://github.com/thewca/worldcubeassociation.org/blob/3eb16bf8669ecb11ac20a89f27f75d83a7380f79/WcaOnRails/app/models/event.rb#L40-L42) method for that.

**Review by jonatanklosko — COMMENTED** [review]
I think it makes sense, although people are using the *Concise*Ranks* from the database export, not sure if that cause any notable issues. @viroulep thoughts?

File: WcaOnRails/app/views/persons/_personal_records.html.erb (hunk 1 above)
Comment (viroulep) [reply WcaOnRails/app/views/persons/_personal_records.html.erb:21]: Additionally, I think this filter should be done earlier: we're not doing anything with the `rank_single` if it's not official.
I'd either:
  - change [these lines](https://github.com/thewca/worldcubeassociation.org/blob/master/WcaOnRails/app/controllers/persons_controller.rb#L32-L33) in the controller to include something like `select { |r| r.event.official? }`
  - create some "sorted_official" scope in [here](https://github.com/thewca/worldcubeassociation.org/blob/2f2a1b591a64fcef5cfb5ad7e0ee01a1e61f4e21/WcaOnRails/app/models/ranks_single.rb#L3) that joins the event table and filter + sort the results. Which then could be used with `@person.rankSingle.sorted_official` I believe (and would remove the need for the `sort_by` above).

Comment (Jambrose777) [reply WcaOnRails/app/views/persons/_personal_records.html.erb:21]: Sure! Those sound good. Does the general idea of removing the exclusion in CAD sound like a good idea?

Comment (viroulep) [reply WcaOnRails/app/views/persons/_personal_records.html.erb:21]: From the top of my head I think so, hopefully we won't find any issues upon completing the next CAD after it's merged :)

Comment (viroulep) [reply WcaOnRails/app/views/persons/_personal_records.html.erb:21]: Oh btw I think the error from travis is the trailing whitespace on this specific line :)

Comment (Jambrose777) [reply WcaOnRails/app/views/persons/_personal_records.html.erb:21]: Iirc the test didn’t pass. I wanted to have the concept approved before fixing things up :D should get around to it this weekend!

Comment (Jambrose777) [reply WcaOnRails/app/views/persons/_personal_records.html.erb]: All Fixed up now :D, the tests that I originally thought was something else

**Review by jonatanklosko — APPROVED** [review]
LGTM

**Comment by jonatanklosko** [comment]
I've just run CAD, looks good!

**Pull request merged**